open-api/rest-catalog-open-api.yaml

```diff
@@ -654,16 +697,26 @@ paths:
     delete:
       tags:
         - Catalog API
-      summary: Uses a `plan-id` as input to cancel a `planTable` operation.
-      operationId: CancelPlan
+      summary: Cancels scan planning for a plan-id
+      operationId: cancelPlanning
       description:
-        Uses a `plan-id` as input to cancel a `planTable` operation.
-        If successful, should return a state that says "cancelled".
-        
-        If an invalid `plan-id` is provided, the service will return a `404 NoSuchPlanIdError` exception indicating that the requested plan does not exist.
+        Cancels scan planning for a plan-id.
+
+
+        This notifies the service that it can release resources for the scan.
+
+
+        Cancellation is not necessary when
+
+        - Scan tasks for each plan task have been fetched using fetchScanTasks
```

Comment:
should we clarify cancellation is not necessary if a `completed` status is returned?

Reply:
Yes, good call.

Reply:
Wait, no. That's actually a case that we specifically want to avoid.

The problem is that there are two times that we may want to cancel a plan-id:
1. Before the initial planning step has completed (what we were calling "pre-planning"), _and_
2. Before all of the plan tasks have been fetched

The idea is that the service may be holding state for plan tasks and if the client is performing a LIMIT query and doesn't consume all of them, it should notify the service that the results of rest of the plan tasks are no longer needed.

I'll clarify this in the doc.